articles/055_ros_parameter_design.md

```diff
@@ -0,0 +1,184 @@
+  The updated values can include unsetting the value.
+  It will provide an API that can atomically update a set of values such that if any of the values fail validation, none of the values are set.
+  The success or failure of calling set parameters will be available to the client.
+- **List Parameters**
+  Provide a list of names of parameters which are currently set.
+- **Describe Parameters**
+  Given a list of parameter names, return their datatype.
+
+This functionality will be exposed through local API calls as well as a ROS Service call API for remote operations.
+
+### Parameter update validation
+
+The node will allow for the registration of a callback for custom parameter value validation.
+
+### Backwards compatibility Parameter Server like behavior
+
+There are use cases where the older behavior with parameters persisting beyond the duration of a specific node are valuable.
+To this end we propose to write a simple node which emulates the policy of the ROS 1.0 parameter server and runs in namespace `/` and simply accepts all change requests.
+
+### Parameter API
+
+The client libraries will provide an API for interfacing with the Core Parameter API for both local and remote nodes including return codes.
+
+### Parameter Change notification
+
+Each node will provide a topic on which parameter changes will be published.
+This topic is to support monitoring parameters for change.
+It is expect that client libraries will implement the ability to register callbacks for specific parameter change notifications using this topic.
+
+### Logging and playback
+
+When logging an entire system the parameter changes can be logged via standard topic recording of the events channel.
+An implementation of the playback mechanism could listen to the parameter event streams and invoke the set parameter calls via the remote API.
+
+## Current implementation
+
+The above specification has been prototyped, the implementation can be found in [rclcpp](https://github.com/ros2/rclcpp).
+The defintion of the services to use for interacting remotely are contained in the [rcl_interfaces package](https://github.com/ros2/rcl_interfaces)
+
+### Unimplemented
+
+Currently there a few parts of the specification unimplemented.
+
+- No parameter subscription registration.
+  The events are published but there is not way to register a callback for changes to a parameter.
+- The ability to register callback to validate parameter updates prior to them being updated is not available.
+- There has been no work on logging and playback of logged parameter changes.
+- The ability to list and get expected validation policy has not been implemented.
+  It is expected to operate at a slightly higher level and possibly be more closely related to the component life cycle.
+
+## Topics not covered in at the moment
+
+There are still topics to discuss and flesh out either in this document or others going forward a few to highlight.
```

Comment:
I don't really understand the last few words of this sentence. Should it be "Going forward, there are still topics to discuss and flesh out either in this document or others. A few to highlight:"